pyccel/codegen/printing/fcode.py

```diff
@@ -2212,15 +2220,30 @@ def _print_ConstructorCall(self, expr):
 
     def _print_NumpyUfuncBase(self, expr):
         type_name = type(expr).__name__
-        func_name = numpy_ufunc_to_fortran[type_name]
+        try:
+            func_name = numpy_ufunc_to_fortran[type_name]
+        except KeyError:
+            errors.report(PYCCEL_RESTRICTION_TODO, severity='fatal')
         code_args = ', '.join(self._print(i) for i in expr.args)
         code = '{0}({1})'.format(func_name, code_args)
         return self._get_statement(code)
 
     def _print_MathFunctionBase(self, expr):
         type_name = type(expr).__name__
-        func_name = math_function_to_fortran[type_name]
-        code_args = ', '.join(self._print(i) for i in expr.args)
+        if type_name == "MathPow":
+            code = self._print_PyccelPow(expr)
+            return code
```

Comment:
Please create a `_print_MathPow` function to specialise rather than adding an if here